src/components/Media/SfwToggle.tsx

```diff
@@ -0,0 +1,64 @@
+import { Popover, Stack, Group, ThemeIcon, Button, Text } from '@mantine/core';
+import { useDisclosure } from '@mantine/hooks';
+import { NextLink } from '@mantine/next';
+import { IconLock } from '@tabler/icons';
+import { useSession } from 'next-auth/react';
+import { useRouter } from 'next/router';
+import React from 'react';
+import { useSfwContext } from './sfwContext';
+import { useSfwStore } from './sfwStore';
+
+export function SfwToggle({ children }: { children: React.ReactElement }) {
+  const { nsfw, type, id } = useSfwContext();
+  const { data: session } = useSession();
```

Comment:
Ditto, let's use `useCurrentUser`
```suggestion
  const user = useCurrentUser();
```